Ghidra/Features/GnuDemangler/src/main/java/ghidra/app/plugin/core/analysis/GnuDemanglerAnalyzer.java

```diff
@@ -84,106 +79,52 @@ public void registerOptions(Options options, Program program) {
 			help,
 			OPTION_DESCRIPTION_USE_KNOWN_PATTERNS);
 
-		options.registerOption(OPTION_NAME_USE_DEPRECATED_DEMANGLER, useDeprecatedDemangler, help,
-			OPTION_DESCRIPTION_DEPRECATED_DEMANGLER);
-
-		options.registerOption(OPTION_NAME_DEMANGLER_PARAMETERS, demanglerParameters, help,
-			OPTION_DESCRIPTION_DEMANGLER_PARAMETERS);
+		options.registerOption(OPTION_NAME_DEMANGLER_FORMAT, demanglerFormat, help,
+			OPTION_DESCRIPTION_DEMANGLER_FORMAT);
+		
+		registerDeprecatedOption(options);
 	}
 
 	@Override
 	public void optionsChanged(Options options, Program program) {
 		doSignatureEnabled = options.getBoolean(OPTION_NAME_APPLY_SIGNATURE, doSignatureEnabled);
 		demangleOnlyKnownPatterns =
 			options.getBoolean(OPTION_NAME_DEMANGLE_USE_KNOWN_PATTERNS, demangleOnlyKnownPatterns);
+		demanglerFormat = options.getEnum(OPTION_NAME_DEMANGLER_FORMAT, GnuDemanglerFormat.AUTO);
 
-		useDeprecatedDemangler =
-			options.getBoolean(OPTION_NAME_USE_DEPRECATED_DEMANGLER, useDeprecatedDemangler);
-
-		demanglerParameters =
-			options.getString(OPTION_NAME_DEMANGLER_PARAMETERS, demanglerParameters);
+		if (demanglerFormat.isDeprecatedFormat() && demanglerFormat.isModernFormat()) {
```

Comment:
So, a couple thoughts here.  This code is a bit hard to follow, which made me question if it is worth even hiding/disabling the checkbox for the 'deprecated' option.

It seems reasonable to me to leave the checkbox always enabled.  It is really just a signal to say whether the user wishes to even attempt to try the old demangler if the modern one fails.  I think it is OK to do nothing if it is selected, the modern one fails and the old demangler cannot handle the format.  In that case, it should just quietly do nothing.

Reply:
I think I'll change the help in this case. The main take away for these changes is that a format isn't attempted to be used on a demangler which doesn't support it. It is also more user friendly.